### PR TITLE
next export details and soft reset

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/sage_intacct/tasks.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 from apps.fyle.models import ExpenseGroup, Reimbursement, Expense
 from apps.tasks.models import TaskLog, Error
 from apps.mappings.models import GeneralMapping
-from apps.workspaces.models import SageIntacctCredential, FyleCredential, Configuration, LastExportDetail
+from apps.workspaces.models import SageIntacctCredential, FyleCredential, Configuration, LastExportDetail, WorkspaceSchedule
 
 from .models import ExpenseReport, ExpenseReportLineitem, Bill, BillLineitem, ChargeCardTransaction, \
```
Comment (Shwetabhk) [inline apps/sage_intacct/tasks.py:25]: can we remove `\` and put everything in parenthesis, 1 import in 1 line?

check how it is done in QBD

File: apps/sage_intacct/tasks.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 from apps.fyle.models import ExpenseGroup, Reimbursement, Expense
 from apps.tasks.models import TaskLog, Error
 from apps.mappings.models import GeneralMapping
-from apps.workspaces.models import SageIntacctCredential, FyleCredential, Configuration, LastExportDetail
+from apps.workspaces.models import SageIntacctCredential, FyleCredential, Configuration, LastExportDetail, WorkspaceSchedule
```
Comment (Shwetabhk) [inline apps/sage_intacct/tasks.py:23]: quite a but if imports in 1 line, lets break it down with parenthesis

**Review by Shwetabhk — APPROVED** [review]

File: apps/mappings/signals.py [hunk 3]
```diff
@@ -88,6 +89,15 @@ def run_post_location_entity_mappings(sender, instance: LocationEntityMapping, *
     workspace.save()
 
 
+@receiver(post_delete, sender=LocationEntityMapping)
+def run_post_delete_location_entity_mappings(sender, instance: LocationEntityMapping, **kwargs):
+    workspace = Workspace.objects.get(id=instance.workspace_id)
+    if workspace.onboarding_state in ('CONNECTION', 'EXPORT_SETTINGS'):
+        DestinationAttribute.objects.filter(~Q(attribute_type='LOCATION'), workspace_id=instance.workspace_id).delete()
```
Comment (ashwin1111) [inline apps/mappings/signals.py:96]: it should be location entity, not location

File: apps/mappings/signals.py [hunk 4]
```diff
@@ -88,6 +89,15 @@ def run_post_location_entity_mappings(sender, instance: LocationEntityMapping, *
     workspace.save()
 
 
+@receiver(post_delete, sender=LocationEntityMapping)
+def run_post_delete_location_entity_mappings(sender, instance: LocationEntityMapping, **kwargs):
+    workspace = Workspace.objects.get(id=instance.workspace_id)
```
Comment (ashwin1111) [inline apps/mappings/signals.py:94]: no need of querying, you can try `instance.workspace.onboarding_state`

File: apps/mappings/views.py [hunk 5]
```diff
@@ -37,6 +37,26 @@ def get(self, request, *args, **kwargs):
...
+                status=status.HTTP_200_OK
+            )
+        
+        except LocationEntityMapping.DoesNotExist:
+            return Response(
+                {
+                    'message': 'Location Entity mappings do not exist for the workspace'
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
```
Comment (ashwin1111) [inline apps/mappings/views.py:58]: just use DestroyView no, can be done in just 1 line

File: apps/workspaces/models.py [hunk 6]
```diff
@@ -196,6 +196,7 @@ class LastExportDetail(models.Model):
     """
     id = models.AutoField(primary_key=True)
     last_exported_at = models.DateTimeField(help_text='Last exported at datetime', null=True)
+    next_export = models.DateTimeField(help_text='next export datetime', null=True)
```
Comment (ashwin1111) [inline apps/workspaces/models.py:199]: anything which involves datetime, let's include suffix `_at`. `next_export_at`

File: apps/mappings/signals.py (hunk 4 above)
Comment (NileshPant1999) [reply apps/mappings/signals.py:94]: we need to chage onboarding state later in the code no ??

**Pull request merged**